### PR TITLE
Add posts benchmarks

### PR DESCRIPTION
**Pull request by TarekkMA**

### CODE REVIEW

File: pallets/posts/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            content: Some(new_content.clone()),
+            space_id: None,
+        };
+    }: update_post(origin, reply.id, update)
+    verify {
+        let updated_post = PostById::<T>::get(reply.id)
+            .ok_or(DispatchError::Other("Post wasn't found"))?;
+
+        ensure!(updated_post != post, "Post wasn't updated");
+        ensure!(updated_post.hidden == true, "Post wasn't updated");
```
Comment (F3Joule) [inline pallets/posts/src/benchmarking.rs:132]:
```suggestion
        ensure!(updated_post.hidden == true, "Post hidden status wasn't updated");
```

File: pallets/posts/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        ensure!(updated_post.content == new_content, "Post wasn't updated");
```
Comment (F3Joule) [inline pallets/posts/src/benchmarking.rs:133]:
```suggestion
        ensure!(updated_post.content == new_content, "Post content wasn't updated");
```

**Review by F3Joule — CHANGES_REQUESTED** [review]

File: pallets/posts/Cargo.toml [hunk 3]
```diff
@@ -12,7 +12,7 @@ categories = ['cryptography::cryptocurrencies']
 
 [features]
 default = ['std']
-runtime-benchmarks = ["frame-benchmarking"]
+runtime-benchmarks = ["frame-benchmarking/runtime-benchmarks"]
 std = [
```
Comment (F3Joule) [inline pallets/posts/Cargo.toml:16]: Add `frame-benchmarking/std` to `std=[...]` to prevent silly errors in the future.

**Review by F3Joule — CHANGES_REQUESTED** [review]

**Comment by F3Joule** [comment]
Also, [this suggestion](https://github.com/dappforce/subsocial-parachain/pull/100#discussion_r1055249266) isn't resolved.

**Review by F3Joule — APPROVED** [review]

File: pallets/posts/Cargo.toml [hunk 4]
```diff
@@ -12,10 +12,12 @@ categories = ['cryptography::cryptocurrencies']
 
 [features]
 default = ['std']
+runtime-benchmarks = ["frame-benchmarking/runtime-benchmarks"]
```
Comment (F3Joule) [inline pallets/posts/Cargo.toml:15]:
```suggestion
runtime-benchmarks = ['frame-benchmarking/runtime-benchmarks']
```

File: pallets/posts/Cargo.toml [hunk 5]
```diff
@@ -26,6 +28,7 @@ std = [
     'pallet-spaces/std',
     'subsocial-support/std',
 ]
+try-runtime = [ "frame-support/try-runtime" ]
```
Comment (F3Joule) [inline pallets/posts/Cargo.toml:31]:
```suggestion
try-runtime = ['frame-support/try-runtime']
```

File: pallets/posts/Cargo.toml [hunk 6]
```diff
@@ -39,6 +42,7 @@ pallet-spaces = { default-features = false, path = '../spaces' }
 subsocial-support = { default-features = false, path = '../support' }
 
 # Substrate dependencies
+frame-benchmarking = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24", default-features = false, optional = true }
```
Comment (F3Joule) [inline pallets/posts/Cargo.toml:45]:
```suggestion
frame-benchmarking = { git = 'https://github.com/paritytech/substrate', branch = 'polkadot-v0.9.24', default-features = false, optional = true }
```

**Review by F3Joule — APPROVED** [review]

**Review by olehmell — APPROVED** [review]

**Pull request merged**